rll/rightscale-mirrors.sh

```
@@ -0,0 +1,300 @@
+  echo "RUBYGEMS_FREEZE_DATE can't be from a date in the future"
+  exit 1
+fi
+
+if [[ -n "$FREEZE_DATE" ]] || [[ -n "$RUBYGEMS_FREEZE_DATE" ]]; then
+  if [[ -z "$MIRROR_HOST" ]]; then
+    echo "MIRROR_HOST cannot be blank, possible values are cf-mirror.rightscale.com or ENV:RS_ISLAND"
+  elif [[ ! "$MIRROR_HOST" =~ island|mirror ]]; then
+    echo "MIRROR_HOST (= $MIRROR_HOST) appears to be invalid, possible values are cf-mirror.rightscale.com or ENV:RS_ISLAND"
+    exit 1
+  fi
+fi
+
+distro=unknown
+distro_ver=unknown
+if which apt-get >/dev/null 2>&1; then
+  if [[ -f /etc/lsb-release ]]; then
+    . /etc/lsb-release
+    distro=${DISTRIB_ID:-unknown}
+    distro_ver=${DISTRIB_RELEASE:-unknown}
+    distro_codename=${DISTRIB_CODENAME:-unknown}
+  fi
+elif [[ -f /etc/centos-release ]]; then
+  if [[ $(cat /etc/centos-release) =~ ^([^0-9]+)\ ([0-9])\. ]]; then
+    distro="${BASH_REMATCH[1]}"
+    distro=${distro/Red Hat/RedHat}
+    distro_ver="${BASH_REMATCH[2]}"
+    distro_ver=$(echo $distro_ver | cut -d. -f1)
+  fi
+elif [[ -f /etc/redhat-release ]]; then
+  if [[ $(cat /etc/redhat-release) =~ ^([^0-9]+)\ ([0-9])\. ]]; then
+    distro="${BASH_REMATCH[1]}"
+    distro=${distro/Red Hat/RedHat}
+    distro_ver="${BASH_REMATCH[2]}"
+    distro_ver=$(echo $distro_ver | cut -d. -f1)
+  fi
+fi
+distro=$(echo $distro | cut -d' ' -f1 | tr [:upper:] [:lower:])
+arch_info=$(echo $(uname -i 2>/dev/null)$(uname -m 2>/dev/null))
+if [[ arch_info =~ 386|686 ]]; then
+  arch=i386
+else
+  arch=x86_64
+fi
+
+freeze_date_msg=$FREEZE_DATE
+if [[ -z "$FREEZE_DATE" ]]; then
+  freeze_date_msg='(unconfigured)'
+fi
+rubygems_freeze_date_msg=$RUBYGEMS_FREEZE_DATE
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  rubygems_freeze_date_msg='(unconfigured)'
+fi
+echo "Distro: $distro"
+echo "Distro version: $distro_ver"
+echo "Architecture: $arch"
+echo "OS Repositories Freeze Date: $freeze_date_msg"
+echo "Rubygems Freeze Date: $RUBYGEMS_FREEZE_DATE"
```

Should we fail if any of these variables are not set correctly?  I don't see any failure catches in the conditions above.